sklearn/inspection/_permutation_importance.py

```diff
@@ -143,14 +167,39 @@ def permutation_importance(estimator, X, y, *, scoring=None, n_repeats=5,
     random_state = check_random_state(random_state)
     random_seed = random_state.randint(np.iinfo(np.int32).max + 1)
 
-    scorer = check_scoring(estimator, scoring=scoring)
-    baseline_score = _weights_scorer(scorer, estimator, X, y, sample_weight)
-
-    scores = Parallel(n_jobs=n_jobs)(delayed(_calculate_permutation_scores)(
-        estimator, X, y, sample_weight, col_idx, random_seed, n_repeats, scorer
-    ) for col_idx in range(X.shape[1]))
-
-    importances = baseline_score - np.array(scores)
-    return Bunch(importances_mean=np.mean(importances, axis=1),
-                 importances_std=np.std(importances, axis=1),
-                 importances=importances)
+    if scoring is None or isinstance(scoring, str) or callable(scoring):
+        scorer = check_scoring(estimator, scoring=scoring)
+        baseline_score = _weights_scorer(scorer, estimator, X, y,
+                                         sample_weight)
+
+        scores = Parallel(n_jobs=n_jobs)(
+            delayed(_calculate_permutation_scores)(
+                estimator, X, y, sample_weight, col_idx, random_seed,
+                n_repeats, scorer
+            ) for col_idx in range(X.shape[1]))
+
+        importances = baseline_score - np.array(scores)
```

Comment:
We could define a small helper function that given 2 arrays, baseline and the permuted scores respectively, will compute the importances and return the Bunch.

In this way we could have

```python
if isinstance(baseline_score, dict):
    # return a dict for each score requested
    return {
        name: _create_importances_bunch(baseline_score[name], scores[name])
        for name in baseline_score
    }
else:
    return _create_importances_bunch(baseline_score, scores)
```

Reply:
done but still need to unpack the scores before for multiscorers